pkg/deploy/api/types.go

```diff
@@ -245,27 +245,46 @@ const DeploymentCancelledAnnotationValue = "true"
 type DeploymentConfig struct {
 	unversioned.TypeMeta
 	kapi.ObjectMeta
+
+	// Spec represents a desired deployment state and how to deploy to it.
+	Spec DeploymentConfigSpec
+
+	// Status represents the current deployment state.
+	Status DeploymentConfigStatus
+}
+
+// DeploymentConfigSpec represents the desired state of the deployment.
+type DeploymentConfigSpec struct {
+	// Strategy describes how a deployment is executed.
+	Strategy DeploymentStrategy
+
 	// Triggers determine how updates to a DeploymentConfig result in new deployments. If no triggers
 	// are defined, a new deployment can only occur as a result of an explicit client update to the
 	// DeploymentConfig with a new LatestVersion.
 	Triggers []DeploymentTriggerPolicy
-	// Template represents a desired deployment state and how to deploy it.
-	Template DeploymentTemplate
+
+	// Replicas is the number of desired replicas.
+	Replicas int
+
+	// Selector is a label query over pods that should match the Replicas count.
+	Selector map[string]string
```

Comment:
> The new fields were behind DeploymentTemplate previously.

Ah, definitely interested in that separate pull then.  Checking congruence in one and then reasoning about changes in another..

Couldn't we have updated the `Selector` of the DC and orphaned existing pods for an old RC created by this DC?